src/render.js

```diff
@@ -6,9 +6,15 @@ module.exports = function render() {
   if (!mapExists) return cleanup();
 
   var mode = store.ctx.events.currentModeName();
-  store.ctx.ui.queueMapClasses({
-    mode: mode
-  });
+
+  // If we were in direct select mode and our selected feature no longer exists
+  // (because it was deleted), we need to get out of that mode.
+  if (mode === Constants.modes.DIRECT_SELECT && !store.get(store.getSelectedIds()[0])) {
```

Comment:
hmmm... this is ok for now, but I'd like to find a way to keep mode logic out of the renderer. Maybe there is a way to emit events only to the modes when its an application triggered event. This way modes could listen to the delete and selection change events and update themselves.